google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITBuiltInMetricsTest.java

```diff
@@ -0,0 +1,117 @@
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Duration;
+import org.threeten.bp.Instant;
+
+@RunWith(JUnit4.class)
+public class ITBuiltInMetricsTest {
+
+  private static Database db;
+  @ClassRule public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  private static DatabaseClient client;
+
+  private static MetricServiceClient metricClient;
+
+  @BeforeClass
+  public static void setUp() throws IOException {
+    metricClient = MetricServiceClient.create();
+    env.getTestHelper().getOptions().toBuilder().setEnableBuiltInMetrics(true);
+    db = env.getTestHelper().createTestDatabase();
+    client = env.getTestHelper().getDatabaseClient(db);
+  }
+
+  @Test
+  public void testBuiltinMetricsWithDefaultOTEL() throws Exception {
+    // This stopwatch is used for to limit fetching of metric data in verifyMetrics
+    Stopwatch metricsPollingStopwatch = Stopwatch.createStarted();
+    Instant start = Instant.now().minus(Duration.ofMinutes(10));
+    Instant end = Instant.now().plus(Duration.ofMinutes(3));
+    ProjectName name = ProjectName.of(env.getTestHelper().getOptions().getProjectId());
+
+    TimeInterval interval =
+        TimeInterval.newBuilder()
+            .setStartTime(Timestamps.fromMillis(start.toEpochMilli()))
+            .setEndTime(Timestamps.fromMillis(end.toEpochMilli()))
+            .build();
+    String ddl =
+        "CREATE TABLE FOO ("
+            + "  K    STRING(MAX) NOT NULL,"
+            + "  V    INT64,"
+            + ") PRIMARY KEY (K)";
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op =
+        db.updateDdl(Collections.singletonList(ddl), null);
+    op.get();
+
+    client
+        .readWriteTransaction()
+        .run(transaction -> transaction.executeQuery(Statement.of("Select 1")));
+
+    String metricFilter =
+        String.format(
+            "metric.type=\"spanner.googleapis.com/client/%s\" "
+                + "AND resource.labels.instance=\"%s\" AND metric.labels.method=\"Spanner.ExecuteStreamingSql\""
+                + " AND metric.labels.database=\"%s\"",
+            "operation_latencies", env.getTestHelper().getInstanceId(), db.getId());
+
+    ListTimeSeriesRequest.Builder requestBuilder =
+        ListTimeSeriesRequest.newBuilder()
+            .setName(name.toString())
+            .setFilter(metricFilter)
+            .setInterval(interval)
+            .setView(ListTimeSeriesRequest.TimeSeriesView.FULL);
+
+    ListTimeSeriesRequest request = requestBuilder.build();
+
+    ListTimeSeriesResponse response = metricClient.listTimeSeriesCallable().call(request);
+    while (response.getTimeSeriesCount() == 0
+        && metricsPollingStopwatch.elapsed(TimeUnit.MINUTES) < 10) {
```

Comment:
It does not sound reasonable that a test like this should be allowed to run for 10 minutes while waiting for results. Can we fail earlier?

Reply:
I have changed the time from 10 minutes to 3 minutes. Rationale is that we are getting the timeseries data from the server and it can take some time for the data to reach there (delay in exporter, or delay in precomputation etc ) .

Note: I will optimize this time once I am able to test it thoroughly after backend changes are available.